src/CMakeLists.txt

```diff
@@ -839,9 +846,13 @@ if(WITH_LIBCEPHSQLITE)
   set(cephsqlite_srcs libcephsqlite.cc SimpleRADOSStriper.cc)
   add_library(cephsqlite ${CEPH_SHARED} ${cephsqlite_srcs})
   target_link_libraries(cephsqlite PRIVATE cls_lock_client librados ceph-common SQLite3::SQLite3 ${EXTRALIBS})
-  set_target_properties(cephsqlite PROPERTIES
-    CXX_VISIBILITY_PRESET hidden
-    VISIBILITY_INLINES_HIDDEN ON)
+  if(NOT APPLE)
+    # clang doesn't like the combination of these properties
```

Comment:
Clang or Apple-clang?

Reply:
Eventually, I had to get to the bottom of this, cause the thing began crashing into my face. The findings are concrete but I still can't say I understand it fully.
It turns out, that clang treats "hidden" "static" "inline" "thread_local" members in a way that causes the duplicate symbol error. I have been able to come up with a minimal reproduction.
I found a reasonable workaround for this particular case, making sure that the said member is explicitly marked as visibility=default
```
#if __clang__
  // clang will generate conflicting symbols
  // when the same inline static thread_local variable
  // is accessed from multiple CUs (access in this header
  // is effectively access from every CU including it)
  // while having visibility = hidden
  // IDK if it's a bug or a feature, but definitely
  // different from how GCC handles it.
  // Some targets in ceph (namely, cephsqlite)
  // are compiled with the default visibility 'hidden',
  // so this declaration will counter that setting
  // and prevent a "duplicate symbol" error during linking
  __attribute__((visibility("default")))
#endif
  inline static thread_local Cache cache;
```